src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -187,16 +187,46 @@ def _create_pixel_coordinate_dataframe(
             [y for y in range(0, data.shape[2])] for x in range(0, data.shape[1])
         ]
 
-        rd_x, rd_y = rasterio.transform.xy(
-            self.dataset.transform, x_coordinates, y_coordinates
+        rd_x_ul, rd_y_ul = rasterio.transform.xy(
```

Comment:
Suggestion: Maybe a small comment here what the 'ul', 'ur', 'll', 'lr' abbreviations mean? Because I'm not really sure right now.